nck/utils/date_handler.py

```diff
@@ -40,18 +40,16 @@ def __get_previous_month_dates(current_date: date) -> Tuple[date, date]:
 
 
 def check_date_range_definition_conformity(start_date: date, end_date: date, date_range: str):
-    if (start_date is None and end_date is None) and date_range is None:
-        raise NoDateDefinitionException(
-            "You must at least define a couple \
-                        start-date/end-date or a date-range"
-        )
-    elif (start_date is None or end_date is None) and (start_date is not None or end_date is not None):
-        raise MissingDateDefinitionException("If you define dates, both start_date and end_date must be defined")
-    if start_date is not None and end_date is not None and date_range is not None:
-        raise InconsistentDateDefinitionException(
-            "You must define either start_date and end_date \
-            or date_range, but not both"
-        )
+    if bool(start_date) != bool(end_date):
+        raise MissingDateDefinitionException("Both start_date and end_date must be defined")
+    if date_range:
+        if all([start_date, end_date]):
+            raise InconsistentDateDefinitionException("You must define either (start_date, end_date) or date_range")
+    else:
+        if not any([start_date, end_date]):
+            raise NoDateDefinitionException("You must at least define a couple (start-date, end-date) or a date-range")
+        elif end_date < start_date:
+            raise InconsistentDateDefinitionException("Report end date should be equal or ulterior to report start date.")
```

Comment:
Shouldn't it rather be:
```
def check_date_range_definition_conformity(start_date: date, end_date: date, date_range: str):
    if date_range:
        if any([start_date, end_date]):
            raise DateDefinitionException("You must define either (start_date, end_date) or date_range")
    else:
        if not all([start_date, end_date]):
            raise DefinitionException("You must at least define a couple (start-date, end-date) or a date-range")
        elif end_date < start_date:
            raise DateDefinitionException("Report end date should be equal or ulterior to report start date.")
```

Reply:
Yes, looks better indeed. I got confused 

Otherwise I can't change the Exception names in this PR because they are used in other readers. We should create another PR t fix it.

Reply:
Actually I added a new generic DateDefinitionException for these checks